server/apps/immich/src/api-v1/asset/asset.service.ts

```diff
@@ -505,6 +541,16 @@ export class AssetService {
     }
   }
 
+  private async checkUserAccess(authUser: AuthUserDto, userId: string) {
+    // Check if userId shares assets with authUser
+    const hasAccess = (await this.partnerCore.getAll(authUser.id, 'shared-with')).some(
+      (item) => item.sharedBy == userId,
```

Comment:
It'd be great to filter this in the database itself, rather than loading it into memory and then filtering.

Reply:
Ah, sure. `PartnerCore` already has the needed method